main/src/cgeo/geocaching/utils/ClipboardUtils.java

```diff
@@ -25,22 +26,23 @@ private ClipboardUtils() {
      * @param text
      *            The text to place in the clipboard.
      */
-    @SuppressWarnings("deprecation")
     public static void copyToClipboard(@NonNull final CharSequence text) {
         // fully qualified name used here to avoid buggy deprecation warning (of javac) on the import statement
-        final android.text.ClipboardManager clipboard = (android.text.ClipboardManager) CgeoApplication.getInstance().getSystemService(Context.CLIPBOARD_SERVICE);
-        clipboard.setText(text);
+        final ClipboardManager clipboard = (ClipboardManager) CgeoApplication.getInstance().getSystemService(Context.CLIPBOARD_SERVICE);
+        if (clipboard != null) {
+            final ClipData data = ClipData.newPlainText(null, text);
+            clipboard.setPrimaryClip(data);
+        }
     }
 
     /**
      * get clipboard content
      *
      */
-    @SuppressWarnings("deprecation")
     @Nullable
     public static String getText() {
         // fully qualified name used here to avoid buggy deprecation warning (of javac) on the import statement
-        final android.text.ClipboardManager clipboard = (android.text.ClipboardManager) CgeoApplication.getInstance().getSystemService(Context.CLIPBOARD_SERVICE);
+        final ClipboardManager clipboard = (ClipboardManager) CgeoApplication.getInstance().getSystemService(Context.CLIPBOARD_SERVICE);
         final CharSequence text = clipboard.getText();
```

Comment:
`clipboard.getText();` is also deprecated. Is there a chance to find a solution as fefore for `setText()`?

Reply:
Perhaps, though I'm not certain about how to go about it, so I'm leaving it as is so I don't break anything by accident.